k9mail-library/src/main/java/com/fsck/k9/mail/Address.java

```diff
@@ -302,16 +303,17 @@ public static String quoteAtoms(final String text) {
      * double quote character to start and end if it's not already there.
      * sample -> "sample"
      * "sample" -> "sample"
-     * ""sample"" -> "sample"
+     * ""sample"" -> ""sample""
      * "sample"" -> "sample"
      * sa"mp"le -> "sa"mp"le"
      * "sa"mp"le" -> "sa"mp"le"
      * (empty string) -> ""
-     * " -> ""
+     * " -> """
      * @param s
      * @return
      */
-    private static String quoteString(String s) {
+    @VisibleForTesting
+    public static String quoteString(String s) {
```

Comment:
Package private should always be enough for testing. Use the same package for tests.